src/components/Button/Button.tsx

```diff
@@ -60,30 +63,38 @@ const Button = forwardRef<HTMLButtonElement | HTMLAnchorElement, ButtonProps>(
       isDisabled = false,
       href,
       target,
+      prefetch = 'none',
       ...props
     },
     ref,
   ) => {
     const { className: baseClassName, ...commonProps } = getCommonProps(props, 'Button');
+    const [linkHovered, setLinkHovered] = useState(false);
     if (href) {
       return (
-        <a
-          {...commonProps}
-          ref={ref as React.ForwardedRef<HTMLAnchorElement>}
-          href={href}
-          className={classnames(
-            `${baseClassName}`,
-            `${baseClassName}--${variant}`,
-            {
-              [`${baseClassName}--icon-last`]: iconLast,
-            },
-            className,
-          )}
-          target={target}
-          rel={target === '_blank' ? 'noopener noreferrer' : undefined}
-        >
-          {children}
-        </a>
+        <>
+          <a
+            {...commonProps}
+            ref={ref as React.ForwardedRef<HTMLAnchorElement>}
+            href={href}
+            className={classnames(
+              `${baseClassName}`,
+              `${baseClassName}--${variant}`,
+              {
+                [`${baseClassName}--icon-last`]: iconLast,
+              },
+              className,
+            )}
+            target={target}
+            rel={target === '_blank' ? 'noopener noreferrer' : undefined}
+            onMouseOver={() => setLinkHovered(true)}
+            onClick={props.onClick}
+          >
+            {children}
+          </a>
+          {prefetch === 'intent' && linkHovered && <link data-testid="prefetch-link" rel="prefetch" href={href} />}
```

Comment:
I think there's a second link added to preload the modules: https://remix.run/docs/en/main/components/prefetch-page-links
Since you're sort of reproducing that code